py-polars/polars/eager/frame.py

```diff
@@ -196,130 +203,189 @@ def __init__(
             nullable = columns
             columns = None
 
-        # Parse data into a list of Series
-        data_series: tp.List["pl.Series"]
-
         if data is None:
-            data_series = []
+            self._df = dict_to_pydf({}, columns=columns, nullable=nullable)
 
         elif isinstance(data, dict):
-            data_series = [
-                pl.Series(k, v, nullable=nullable).inner() for k, v in data.items()
-            ]
+            self._df = dict_to_pydf(data, columns=columns, nullable=nullable)
 
         elif isinstance(data, np.ndarray):
```

Comment:
This is great. Greatly improves readability by dispatching to specialized functions.